typescript-decorators/src/controllers/main.controller.ts

```diff
@@ -0,0 +1,59 @@
+import { FastifyReply, FastifyRequest } from 'fastify'
+import { AbstractController, Controller, ControllerType, GET, Hook, POST } from 'fastify-decorators';
+import { IncomingMessage, ServerResponse } from 'http'
+
+// Define controller
+@Controller({
+    route: '/main', // Base URL for all controller handlers
+    type: ControllerType.SINGLETON // SINGLETON is default controller type, just define it explicit
+})
+class MainController
+    // It's optional. We have to extends AbstractController to have Fastify instance accessible inside.
+    extends AbstractController {
+    public get message(): string {
+        return this._message;
+    }
+
+    public set message(value: string) {
+        // Using Fastify instance to log that field was changed
+        this.instance.log.info(`New value received: ${value}`);
```

Comment:
@mcollina, friendly ping!

That's it!

Reply:
In near future I'm gonna implement DI.
There is the idea about how it would looks like: https://gist.github.com/L2jLiga/090064af76b046153eb7c448fcf69bfb